x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/user/TransportSetEnabledAction.java

```diff
@@ -62,4 +64,16 @@ protected void doExecute(Task task, SetEnabledRequest request, ActionListener<Ac
             listener.delegateFailure((l, v) -> l.onResponse(ActionResponse.Empty.INSTANCE))
         );
     }
+
+    private boolean isSameUserRequest(SetEnabledRequest request) {
+        final var effectiveSubject = securityContext.getAuthentication().getEffectiveSubject();
```

Comment:
I'm matching the existing logic (aside from the realm improvement), however there is another possible case to prevent: currently, nothing keeps alice from running as bob to disable alice. I can extend the logic here to account for that but not sure if worth it.

Reply:
Makes sense, thanks for the context!